### PR TITLE
GitHub action to run periodically or by trigger to extract configs

### LINKED ISSUE #143: GitHub action to run periodically or by trigger to extract configs
For now we just need to run `./gradlew build` and commit and push to main branch all channels.

But this can be done as GitHub action, to be done daily.

Example on how to do this:
https://github.com/checkstyle/checkstyle/blob/977931a4c52791d55adce728106eef3766502e12/.github/workflows/release-maven-prepare.yml#L7-L20

https://github.com/checkstyle/checkstyle/blob/977931a4c52791d55adce728106eef3766502e12/.github/workflows/release-maven-prepare.yml#L43-L47

https://github.com/checkstyle/checkstyle/blob/977931a4c52791d55adce728106eef3766502e12/.github/workflows/release-maven-prepare.yml#L52-L54

### PR DESCRIPTION
**Pull request by relentless-pursuit**
Resolves #143

### CODE REVIEW

**Comment by relentless-pursuit** [comment]
Proof of work:
image

https://github.com/checkstyle-gsoc/test-configs/commit/266d17fbb75386c390b3240e24d72176e655f2cd
https://github.com/checkstyle-gsoc/test-configs/actions/runs/10640612696

@Romani please review

**Comment by relentless-pursuit** [comment]
@romani 
Can we merge this and trigger it to see the automated addition? I know the ci build will fail complaining of the git diff issue. But, i was thinking if we don't create a PR rather use this workflow to do that.

**Review by romani — APPROVED** [review]

File: .github/workflows/automate.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      run: ./setup.sh
+
+    - name: Build and Test with Gradle
+      run: |
+        cd extractor
+        ./gradlew clean build test run --args="../.ci-temp/checkstyle" --info --stacktrace --console=plain \
+          -Dorg.gradle.logging.level=DEBUG \
+          -Dtest.single=MainsLauncherTest
+
+    - name: Check for changes
```
Comment (romani) [inline .github/workflows/automate.yml:31]: Remove this step, it is not required

File: .github/workflows/automate.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+name: Java CI
```
Comment (romani) [inline .github/workflows/automate.yml:1]: Rename it to cron-extract.yml

File: .github/workflows/automate.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+on: [push, pull_request]
```
Comment (romani) [inline .github/workflows/automate.yml:3]: Put cron execution instead, execute ones a day

Comment (romani) [reply .github/workflows/automate.yml:3]: Add ability to be execute by manual trigger.

File: .github/workflows/automate.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        chmod +x ./.github/workflows/workflow-command.sh
+        ./.github/workflows/workflow-command.sh git_check_changes
+
+    - name: Commit changes
+      if: steps.git-check.outputs.changes == 'true'
+      run: |
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub Action"
+        git add .
+        git commit -m "Automated changes" -a
```
Comment (romani) [inline .github/workflows/automate.yml:43]: Update from extractor auto execution

**Review by romani — COMMENTED** [review]
Please fix in next PR

File: .github/workflows/automate.yml (hunk 1 above)
Comment (relentless-pursuit) [reply .github/workflows/automate.yml:31]: how would it then find changes?

Comment (romani) [reply .github/workflows/automate.yml:31]: No need to do this just commit all changes

**Pull request merged**